### PR TITLE
feat: add `CNCF Distribution` module

### PR DESCRIPTION
**Pull request by esposm03**
This PR adds a module for CNCF Distrubution, a registry for container images. Some basic testing is included.

### CODE REVIEW

File: src/cncf_distribution/mod.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    fn name(&self) -> String {
+        NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        TAG.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::millis(500)]
```
Comment (DDtKey) [inline src/cncf_distribution/mod.rs:40]: Is there any way to depend on stdout/stderr? It's much better, time is too flaky condition usually

**Review by DDtKey — COMMENTED** [review]
Thanks for the contribution!

File: src/cncf_distribution/mod.rs (hunk 1 above)
Comment (esposm03) [reply src/cncf_distribution/mod.rs]: Yes, I have just switched it from `millis` to `message_on_stderr`

File: src/cncf_distribution/Dockerfile.test
Comment (DDtKey) [inline src/cncf_distribution/Dockerfile.test:1]: Sorry, missed this one

I'd prefer not to keep this file under the `src` directory 

In general it's so small that we can keep it hardcoded as constant string in tests

Comment (esposm03) [reply src/cncf_distribution/Dockerfile.test:1]: Alright, I have just pushed a fix for this, by inlining the Dockerfile inside the source file.

**Comment by mervyn-mccreight** [comment]
Can you resolve the existing conflicts please? I'd integrate this into the next release then :)

**Comment by esposm03** [comment]
Done👍

**Review by mervyn-mccreight — APPROVED** [review]

**Pull request merged**